[nvbugs/5369799] fix: Update disaggregation handling in sampler

stnie opened the pull request:
- Fixed related bug, where sequence length buffer of decoder State was not correctly updated, when a request is passed between processes in disaggregated Serving. This previously lead to the first token being dropped from the output.
- Adjusted the condition, when to add token, to no longer expect the above behavior. The previous condition caused the duplication of the last token under specific cirumstances, which is now also solved (See nvbugs 5369799)
## Description

Full fix for [nvbugs/5369799] : 
Using Disaggregated Serving + overlap scheduler now correctly does not drop an output token, which allows us to remove the Workaround 
As a result: Using Disaggregated Serving, Overlap scheduler and Stop words, now correctly no longer outputs the last token twice.


## Test Coverage

<!--
Please list clearly what are the relevant test(s) that can safeguard the changes in the PR. This helps us to ensure we have sufficient test coverage for the PR.
-->

## GitHub Bot Help

`/bot [-h] ['run', 'kill', 'skip', 'reuse-pipeline'] ...`

Provide a user friendly way for developers to interact with a Jenkins server.

Run `/bot [-h|--help]` to print this help message.

See details below for each supported subcommand.

<details>

`run  [--disable-fail-fast --skip-test --stage-list "A10-1, xxx" --gpu-type "A30, H100_PCIe" --add-multi-gpu-test --only-multi-gpu-test --disable-multi-gpu-test --post-merge --extra-stage "H100_PCIe-[Post-Merge]-1, xxx"]`

Launch build/test pipelines. All previously running jobs will be killed.

`--disable-fail-fast ` *(OPTIONAL)* : Disable fail fast on build/tests/infra failures.

`--skip-test ` *(OPTIONAL)* : Skip all test stages, but still run build stages, package stages and sanity check stages. Note: Does **NOT** update GitHub check status.

`--stage-list "A10-1, xxx"` *(OPTIONAL)* : Only run the specified test stages. Examples: "A10-1, xxx". Note: Does **NOT** update GitHub check status.

`--gpu-type "A30, H100_PCIe"` *(OPTIONAL)* : Only run the test stages on the specified GPU types. Examples: "A30, H100_PCIe". Note: Does **NOT** update GitHub check status.

`--only-multi-gpu-test ` *(OPTIONAL)* : Only run the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--disable-multi-gpu-test ` *(OPTIONAL)* : Disable the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--add-multi-gpu-test ` *(OPTIONAL)* : Force run the multi-GPU tests. Will also run L0 pre-merge pipeline.

`--post-merge ` *(OPTIONAL)* : Run the L0 post-merge pipeline instead of the ordinary L0 pre-merge pipeline.

`--extra-stage "H100_PCIe-[Post-Merge]-1, xxx"` *(OPTIONAL)* : Run the ordinary L0 pre-merge pipeline and specified test stages. Examples: --extra-stage "H100_PCIe-[Post-Merge]-1, xxx".

For guidance on mapping tests to stage names, see `docs/source/reference/ci-overview.md`.

### kill

`kill  `

Kill all running builds associated with pull request.

### skip

`skip --comment COMMENT `

Skip testing for latest commit on pull request. `--comment "Reason for skipping build/test"` is required. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

### reuse-pipeline

`reuse-pipeline `

Reuse a previous pipeline to validate current commit. This action will also kill all currently running builds associated with the pull request. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

</details>


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Bug Fixes**
  * Improved accuracy in calculating current sequence lengths for decoder requests, ensuring more precise token management during generation.
  * Refined logic for detecting and processing new tokens in both single-beam and multi-beam scenarios, removing special handling for certain overlap conditions.

* **Style**
  * Removed unnecessary blank lines to improve code readability.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

stnie committed:
fix: Update disaggregation handling in sampler and executor

- Set `is_disagg` in `TRTLLMSampler` to False by default.
- Modify sequence length calculation to consider `is_disagg` when determining overlap behavior in the sampling process.
- Ensure `kv_cache_transceiver` presence updates `sampler.is_disagg` in the executor instance creation.

Signed-off-by: Stefan Niebler <82932102+stnie@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #11013](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/11013/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #11013](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/11013/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #8138](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/8138/) completed with status: 'SUCCESS'
Pipeline passed with automatic retried tests. Check the [rerun report](https://urm.nvidia.com/artifactory/sw-tensorrt-generic/llm-artifacts//LLM/main/L0_MergeRequest_PR/8138/test-results/rerun_report.html) for details.

stnie committed:
[fix] Fix bug 5369799: Beam search with disaggregated generation.

Fixed Disaggregated Serving + overlap scheduler dropping tokens in the output.
CopySequenceLengths now correctly respects tokens generated in the first generation phase.
Removed the line in sampler.py that masked this issue and caused bug 5367999

Signed-off-by: Stefan Niebler <82932102+stnie@users.noreply.github.com>

stnie commented:
/bot run

stnie committed:
chore: removed unused code lines.

Signed-off-by: Stefan Niebler <82932102+stnie@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #12196](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/12196/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #12196](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/12196/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #9059](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/9059/) completed with status: 'FAILURE'

tensorrt-cicd commented:
[PR_Github #12211](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/12211/) [ run ] triggered by Bot

stnie commented:
/bot run

tensorrt-cicd commented:
[PR_Github #12217](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/12217/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #12211](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/12211/) [ run ] completed with state `ABORTED`

tensorrt-cicd commented:
[PR_Github #12217](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/12217/) [ run ] completed with state `FAILURE`
[/LLM/main/L0_MergeRequest_PR pipeline #9072](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/9072/) completed with status: 'FAILURE'

stnie commented:
/bot run

tensorrt-cicd commented:
[PR_Github #12294](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/12294/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #12294](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/12294/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #9130](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/9130/) completed with status: 'SUCCESS'
